app/javascript/lib/metrics-api.js

```diff
@@ -0,0 +1,277 @@
+  'staging': 'https://terra-bard-alpha.appspot.com',
+  'production': 'https://terra-bard-prod.appspot.com'
+}
+let bardDomain = ''
+let env = ''
+let userId = ''
+
+// TODO (SCP-2237): Use Node environment to get React execution context
+if ('SCP' in window) {
+  env = window.SCP.environment
+  bardDomain = bardDomainsByEnv[env]
+  // To consider: Replace SCP-specific userId with DSP-wide userId
+  userId = window.SCP.userId
+}
+
+/**
+ * Log page view, i.e. page load
+ */
+export function logPageView() {
+  log('page:view')
+}
+
+/** Log click on page.  Delegates to more element-specific loggers. */
+export function logClick(event) {
+  const target = event.target
+  const tag = target.localName.toLowerCase() // local tag name
+
+  if (tag === 'a') {
+    logClickLink(target)
+  } else if (tag === 'button') {
+    logClickButton(target)
+  } else if (tag === 'input') {
+    logClickInput(target)
+  } else {
+    // Perhaps uncomment when Mixpanel quota increases
+    // logClickOther(target)
+  }
+}
+
+/**
+ * Log click on link, i.e. anchor (<a ...) tag
+ */
+function logClickLink(target) {
+  const props = { text: target.text }
+  log('click:link', props)
+}
+
+/**
+ * Log click on button, e.g. for pagination, "Apply", etc.
+ */
+function logClickButton(target) {
+  const props = { text: target.text }
+  log('click:button', props)
+
+  // Google Analytics fallback: remove once Bard and Mixpanel are ready for SCP
+  ga('send', 'event', 'click', 'button')
+}
+
+/**
+ * Get label elements for an input element
+ *
+ * From https://stackoverflow.com/a/15061155
+ */
+function getLabelsForInputElement(element) {
+  let labels
+  const id = element.id
+
+  if (element.labels) {
+    return element.labels
+  }
+
+  if (id) {
+    labels = Array.from(document.querySelector(`label[for='${id}']`))
+  }
+
+  while (element = element.parentNode) {
+    if (element.tagName.toLowerCase() == 'label') {
+      labels.push(element)
+    }
+  }
+
+  return labels
+};
+
+/**
+ * Log click on input by type, e.g. text, number, checkbox
+ */
+function logClickInput(target) {
+  const domLabels = getLabelsForInputElement(target)
+
+  // User-facing label
+  const label = domLabels.length > 0 ? domLabels[0].innerText : ''
+
+  const props = { label }
+  const element = `input-${target.type}`
+  log(`click:${element}`, props)
+
+  // Google Analytics fallback: remove once Bard and Mixpanel are ready for SCP
+  ga('send', 'event', 'click', element) // eslint-disable-line no-undef
+}
+
+/**
+ * Log clicks on elements that are not otherwise classified
+ */
+function logClickOther(target) {
+  const props = { text: target.text }
+  log('click:other', props)
+
+  // Google Analytics fallback: remove once Bard and Mixpanel are ready for SCP
+  ga('send', 'event', 'click', 'other') // eslint-disable-line no-undef
+}
+
+/**
+ * Count terms, i.e. space-delimited strings, and consider [""] to have 0 terms
+ */
+function getNumberOfTerms(terms) {
```

Comment:
I think this function and the search-related functions below belong in one of the search js files (maybe StudySearchProvider.js ?). I'd like to keep the metrics api free of specific business logic.  The metrics api should just export generic helper methods like 'log' and 'logClick', and then it's up to the components to know how to use them

Reply:
Good point!  Per our chat, we agree that this module is too high level for such SCP-API-specific metrics functions, and that React components are too low level.

So in 483b0701, I moved them to a new module, `scp-api-metrics.js`.